File: src/main/java/com/codeElevate/ServiceBookingSystem/services/client/ClientService.java
package com.codeElevate.ServiceBookingSystem.services.client;

import com.codeElevate.ServiceBookingSystem.dto.AdDTO;
import com.codeElevate.ServiceBookingSystem.dto.AdDetailsForClientDTO;
import com.codeElevate.ServiceBookingSystem.dto.ReservationDTO;
import com.codeElevate.ServiceBookingSystem.dto.ReviewDTO;

import java.util.List;

public interface ClientService {

    List<AdDTO> getAllAds();

    List<AdDTO> searchAdByName(String name);

    boolean bookService(ReservationDTO reservationDTO);

    AdDetailsForClientDTO getAdDetailsByAdId(Long adId);

    List<ReservationDTO> getAllBookingsByUserId(Long userId);

    Boolean giveReview(ReviewDTO reviewDTO);
}

File: src/app/client/services/client.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { UserStoargeService } from 'src/app/basic/services/storage/user-stoarge.service';

const BASIC_URL = "http://localhost:8080/";

@Injectable({
  providedIn: 'root'
})
export class ClientService {

  constructor(private http: HttpClient) { }

  getAllAds(): Observable<any>{
    return this.http.get(BASIC_URL + `api/client/ads`, {
      headers : this.createAuthorizationHeader()
    })
  }

  searchAdByName(name:any): Observable<any>{
    return this.http.get(BASIC_URL + `api/client/search/${name}`, {
      headers : this.createAuthorizationHeader()
    })
  }

  getAdDetailsByAdId(adId:any): Observable<any>{
    return this.http.get(BASIC_URL + `api/client/ad/${adId}`, {
      headers : this.createAuthorizationHeader()
    })
  }

  bookService(bookDTO:any): Observable<any>{
 
    return this.http.post(BASIC_URL + `api/client/book-service`, bookDTO , {
      headers : this.createAuthorizationHeader()
    })
  }

  getMyBookings(): Observable<any>{
    const userId = UserStoargeService.getUserId();
    return this.http.get(BASIC_URL + `api/client/my-bookings/${userId}`,  {
      headers : this.createAuthorizationHeader()
    })
  }

  giveReview(reviewDTO:any): Observable<any>{
    return this.http.post(BASIC_URL + `api/client/review`, reviewDTO , {
      headers : this.createAuthorizationHeader()
    })
  }

  createAuthorizationHeader(): HttpHeaders{
    let authHeaders: HttpHeaders = new HttpHeaders();
    return authHeaders.set(
      'Authorization',
      'Bearer ' + UserStoargeService.getToken()
    )
  }
}
